python/lsst/meas/base/afterburner.py

```diff
@@ -0,0 +1,183 @@
+from collections import namedtuple
+import sys
+import traceback
+
+import lsst.pipe.base
+import lsst.pex.config
+import lsst.daf.base
+
+from .pluginsBase import BasePlugin, BasePluginConfig
+from .pluginRegistry import PluginRegistry
+from .baseLib import FatalAlgorithmError, MeasurementError
+
+# Exceptions that the measurement tasks should always propagate up to their callers
+FATAL_EXCEPTIONS = (MemoryError, FatalAlgorithmError)
+
+__all__ = ("AfterburnerPluginConfig", "AfterburnerPlugin", "AfterburnerConfig", "AfterburnerTask")
+
+class AfterburnerPluginConfig(BasePluginConfig):
+    '''
+    Default configuration class for afterburner plugins
+    '''
+    pass
+
+class AfterburnerPlugin(BasePlugin):
+    '''
+    Base class for after burner plugin
+    '''
+    registry = PluginRegistry(AfterburnerPluginConfig)
+    ConfigClass = AfterburnerPluginConfig
+
+    def __init__(self,config, name, schema, metadata):
+        """!
+        Initialize the afterburner plugin
+
+        @param[in] config     An instance of afterburner config class.
+        @param[in] name       The string the plugin was registered with.
+        @param[in,out] schema The source schema, New fields should be added here to
+                              hold output produced by this plugin.
+        @param[in] metadata   Plugin metadata that will be attached to the output catalog
+        """
+        BasePlugin.__init__(self, config, name)
+
+    @classmethod
+    def getExecutionOrder(cls):
+        ''' Sets the relative order of plugins (smaller numbers run first).
+
+        Afterburner plugins must run with BasePlugin.DEFAULT_AFTERBURNER or higher
+
+        All plugins must implement this method with an appropriate run level
+        '''
+        raise NotImplementedError()
+
+    @property
+    def PlugType(self):
+        '''
+        This defines if the plugin operates on a single source at a time, or expects the whole catalog.
+        The value defaults to single for a single source, set to multi when the plugin expects the whole
+        catalog. If The plugin is of type multi, the fail method should be implemented to accept the whole
+        catalog. If the plugin is of type the fail method should accept a single source record.
```

Comment:
"If the plugin is of type" what?  (I think you're missing a word, "single" probably.)